contents/testing/integration-testing.md

```diff
@@ -0,0 +1,197 @@
+
+After completion of unit testing on individual modules, an **integration test** on `Order Food` and `Order Summary` checks that upon ordering food, the order summary reflects the correct food orders.
+
+After completion of integration testing, a system test on the entire application could mimic a customer's point of view and checks that the functionalities such as `Order Food`, `Payment` and `Generate Report` work as intended.
+
+## Why is integration testing important?
+
+In a software application, each software module is usually designed and unit tested by different programmers.
+Since these programmers may work in isolation and have different understandings of the software requirements, integration testing is necessary to ensure that the software modules work in unity and expose any faults in the interaction between different units.
+
+Suppose the `Order Food` module mentioned earlier allows each user to submit up to 10 orders each time. If the `Order Summary` module can only record a maximum of 5 orders each time, when the orders from `Order Food` are passed to the `Order Summary` module, 5 of the orders may not be taken into account. 
+Hence, there is potential for problems such as such as **data loss**, **inaccurate results** and even **system errors**.
+Integration tests find these discrepancies before any serious damage is done.
+
+Although system testing also help to uncover faults between modules, integration testing is still highly encouraged as it is able to **better isolate** the scope of the features during testing, making it **easier** and **faster** to test.
+In fact, many integration problems can be detected and resolved in the early stage of development, making a developer's life easier to debug when a problem arises in the future.
+More advantages of doing integration testing before system testing can be found [here](https://bitwaretechnologies.com/integration-testing-important-software-testing-life-cycle/).
+
+Hence, integration testing can be useful in the development of any non-trivial software product. 
+Concepts shared in this article can be adopted by different software applications.
+
+On top of that, in the current era of <tooltip content="Practice of merging developers' work continuously into a shared repository">Continuous Integration</tooltip> (CI), integration testing is useful to help with early detecion of defects while maintaining the quality of the product throughout further developments.
+The concept of [Continuous Integration Testing](https://crossbrowsertesting.com/blog/continuous-integration/continuous-integration-testing-delivery/) is also formed to combine integration testing with continuous integration.
+
+:fas-info-circle: More information on instances where integration testing is important can be found in [this article](https://www.guru99.com/integration-testing.html).
```

Comment:
Earlier on, you linked to external articles at the end of the paragraph without an additional info circle. Both ways seem okay, but perhaps you may want to be consistent with how it is presented.